src/Object/Product/Product.php

```diff
@@ -980,4 +977,33 @@ public function isValid()
             return false;
         }
     }
+
+    /**
+     * Makes the alternative images collection unique and removes
+     * the main product image from alt images if present
+     */
+    private function altImagesToUnique()
+    {
+        if ($this->alternateImageUrls->count() == 0) {
+            return;
+        }
+        $images = $this->getAlternateImageUrls();
+        $resetImages = false;
+        $duplicates = array_count_values($images);
```

Comment:
If `images` is null this will throw an error
```suggestion
        $duplicates = $images ? array_count_values($images) : null;
```

Reply:
`$this->getAlternateImageUrls()` returns always an array so there's no need for this one

Reply:
My bad, i was looking at the docblock, but the implementation seems to always return an array.
 https://github.com/Nosto/nosto-php-sdk/blob/develop/src/Types/Product/ProductInterface.php#L188